### PR TITLE
Translate post-07 to Japanese

### PR DESCRIPTION
**Pull request by shimomura1004**
I translated post-07 to Japanese.

@JohnTitor and @woodyZootopia Could you review my translation?

### CODE REVIEW

File: blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+
+![QEMU printing `EXCEPTION: DOUBLE FAULT` because of hardware timer](qemu-hardware-timer-double-fault.png)
+
+ダブルフォルトが発生する理由は、ハードウェアタイマ(正確には [Intel 8253])がデフォルトで有効になっているため、割り込みを有効にするとすぐにタイマ割り込みを受け取り始めるためです。この割り込みのためのハンドラ関数を定義していないため、ダブルフォルトのハンドラが呼ばれることになります。
+
+[Intel 8253]: https://en.wikipedia.org/wiki/Intel_8253
+
+## タイマ割り込みの処理
+
+[上述](#the-8259-pic)した図にある通り、タイマはプライマリの PIC の0番目の線を使います。これはタイマ割り込みは32番(0 + オフセットの32)の割り込みとして CPU に届くということです。32をハードコーディングする代わりに `InterruptIndex` enum に保存することにしましょう:
```
Comment (woodyZootopia) [inline blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md:162]:
```suggestion
[上述](#8259-pic)した図にある通り、タイマはプライマリの PIC の0番目の線を使います。これはタイマ割り込みは32番(0 + オフセットの32)の割り込みとして CPU に届くということです。32をハードコーディングする代わりに `InterruptIndex` enum に保存することにしましょう:
```

File: blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+    print!("k");
+
+    unsafe {
+        PICS.lock()
+            .notify_end_of_interrupt(InterruptIndex::Keyboard.as_u8());
+    }
+}
+```
+
+[上述](#the-8259-pic)した図で見たように、キーボードはプライマリ PIC の1番目の線を使います。これはキーボード割り込みは33番(1 + オフセットの32)の割り込みとして CPU に届くということです。このインデックスを新たな `Keyboard` のヴァリアントとして `InterruptIndex` enum に追加します。enum ヴァリアントの値はデフォルトでは前の値に1を足したもの、すなわち33になるので、値を明示的に指定する必要はありません。割り込みハンドラでは、`k` の文字を表示して割り込みコントローラに EOI 信号を送ります。
```
Comment (woodyZootopia) [inline blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md:557]:
```suggestion
[上述](#8259-pic)した図で見たように、キーボードはプライマリ PIC の1番目の線を使います。これはキーボード割り込みは33番(1 + オフセットの32)の割り込みとして CPU に届くということです。このインデックスを新たな `Keyboard` のヴァリアントとして `InterruptIndex` enum に追加します。enum ヴァリアントの値はデフォルトでは前の値に1を足したもの、すなわち33になるので、値を明示的に指定する必要はありません。割り込みハンドラでは、`k` の文字を表示して割り込みコントローラに EOI 信号を送ります。
```

**Review by woodyZootopia — CHANGES_REQUESTED** [review]
Since the section title has changed from `The 8259 PIC` to `8259 PIC`, you should change the internal link too.

**Comment by woodyZootopia** [comment]
Thank you for PR! Here are quick comments:
* Your article cannot be compiled because of the minor error above. I highly recommend you to install zola and compile the blog yourself (just `zola serve` in the `blog` directory). You can check if there is such a error very easily and visually see your progress :)
* Maybe it would be more readable if you put ` `（半角スペース）around `(` and `)`（半角カッコ）?
  e.g. 33番(1 + オフセットの32)の割り込み -> 33番 (1 + オフセットの32) の割り込み

I just skimmed through the article and am going to take a closer look later.

**Comment by shimomura1004** [comment]
Thank you for your comment! 
I've added spaces around parentheses and checked my translation with zola on my local environment.
I'm very sorry for pushing broken commit.

I look forward to hearing from you :)

By the way, "Zola Check" on github was failed because of dead link in other posts. (I wonder why my local build succeeded...)
Should I fix them with this PR or create another PR?

File: blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md (hunk 2 above)
Comment (shimomura1004) [reply blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md]: I'm very sorry but I'm not familiar with github PR so I created other commit by myself...
Let me revert my commit.

**Comment by JohnTitor** [comment]
> By the way, "Zola Check" on github was failed because of dead link in other posts. (I wonder why my local build succeeded...) Should I fix them with this PR or create another PR?

I'd prefer to create another one so that we don't have to wait for this PR to get that fix merged.

**Review by woodyZootopia — COMMENTED** [review]
Your translation was pretty good overall. I suggested a few possible changes, please take a look.

**Comment by shimomura1004** [comment]
Thank you so much for the review!
I'll check and apply your suggestion tonight.

File: blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+// in src/lib.rs
+
+pub fn hlt_loop() -> ! {
+    loop {
+        x86_64::instructions::hlt();
+    }
+}
+```
+
+`instructions::hlt` 関数はアセンブリ命令の[薄いラッパ][thin wrapper]です。この命令はメモリ安全性を損なわないので安全です。
```
[the remark this answers is not in the document]
Comment (shimomura1004) [reply blog/content/edition-2/posts/07-hardware-interrupts/index.ja.md:455]: ありがとうございます。修正いただいた訳のほうが適切と思いますのでこちらも適用しようと思いますが、"unsafe" の前後にスペースを入れたいと思います。

**Comment by shimomura1004** [comment]
@woodyZootopia
Thank you very much for your kind review! I've applied all your review comment.

@JohnTitor 
> I'd prefer to create another one so that we don't have to wait for this PR to get that fix merged.
I agree with you. Thanks!

**Comment by woodyZootopia** [comment]
Thank you. Review finished on my side. Great work!

**Comment by phil-opp** [comment]
Thanks a lot for this new translation @shimomura1004! And thanks to @woodyZootopia for reviewing!

Don't worry about the zola check error, we can fix that separately.

**Comment by phil-opp** [comment]
Merged manually in https://github.com/phil-opp/blog_os/commit/fe387b5fa6c29fb5287d58322e60ea22512445ca after some squashing. Unfortunately, I forgot to push the squashed history to this PR before pushing the merge commit, so GitHub decided to mark this PR as "closed" instead of "merged" now. I'm sorry about that!

I tried to force-push this branch back to bd15169 but this was no longer permitted.

The commits merged to the `main` branch still have the right author info, of course. See https://github.com/phil-opp/blog_os/commit/00b068a6eb6b551362702af425e38b078c872b2f and https://github.com/phil-opp/blog_os/commit/567ace4f8da87360bfd3e0193837c31860658d94.

**Comment by phil-opp** [comment]
Published at https://os.phil-opp.com/ja/hardware-interrupts/

**Comment by shimomura1004** [comment]
No problem!  Thank you for merging :)

**Pull request closed without merging**